packages/eslint-mdx/src/helpers.ts

```diff
@@ -112,23 +112,34 @@ export const restoreNodeLocation = <T>(
   const start = range[0] + offset
   const end = range[1] + offset
 
+  const restoredStartLine = startLine + startLoc.line
+  const restoredEndLine = startLine + endLoc.line
+
   return Object.assign(node, {
     start,
     end,
     range: [start, end],
     loc: {
       start: {
-        line: startLine + startLoc.line,
-        column: startLoc.column,
+        line: restoredStartLine,
+        column: startLoc.column + (restoredStartLine === 1 ? offset : 0),
       },
       end: {
-        line: startLine + endLoc.line,
-        column: endLoc.column,
+        line: restoredEndLine,
+        column: endLoc.column + (restoredEndLine === 1 ? offset : 0),
       },
     },
   })
 }
 
+export const arrayify = <T, R = T extends Array<infer S> ? S : T>(
+  ...args: T[]
+) =>
```

Comment:
Broadly this seems like a custom implementation of https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat ? :thinking:

Reply:
But also accept non-array to be array.

Reply:
something like:

```ts
export function arrayify<T, R = T extends Array<infer S> ? S : T>(...args: T[]): R {
  return args.flat();
}
```

?

Reply:
But also drop `null/undefined` values.

And `Array#flat` is not available on Node 10

https://node.green/#ES2019-features-Array-prototype--flat--flatMap--Array-prototype-flat

https://github.com/mdx-js/eslint-mdx/blob/master/packages/eslint-plugin-mdx/package.json#L14

Reply:
> And Array#flat is not available on Node 10

And node 10 is nearly EOL https://github.com/nodejs/Release#release-schedule :man_shrugging: 

> But also drop null/undefined values.

```ts
export function arrayify<Parameter, Return = Parameter extends Array<infer Item> ? Item : Parameter>(...args: Parameter[]): Return[] {
  return args.flat().filter((value) => value != null);
}
```

Reply:
> And node 10 is nearly EOL

But that would be a breaking change, I would drop Node 10 support in v2.